bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -263,6 +260,30 @@ where
 		// Decoding failed; allow an empty XCM so the message won't fail entirely.
 		Xcm::new()
 	}
+
+	fn filter_topic(xcm: Xcm<()>) -> (Xcm<()>, Option<[u8; 32]>) {
+		let mut result_xcm = xcm.0;
+
+		// First, try to see if the last instruction is a SetTopic.
+		if !result_xcm.is_empty() {
+			let last_index = result_xcm.len() - 1;
+			if let SetTopic(_) = &result_xcm[last_index] {
+				// We know the last element is a SetTopic. Remove it.
+				if let Some(SetTopic(topic)) = result_xcm.pop() {
+					return (result_xcm.into(), Some(topic));
+				}
+			}
+		}
+
+		// If we didn't find a SetTopic at the end, search for one anywhere in the instructions.
+		if let Some(pos) = result_xcm.iter().rposition(|instr| matches!(instr, SetTopic(_))) {
+			if let SetTopic(topic) = result_xcm.remove(pos) {
+				return (result_xcm.into(), Some(topic));
+			}
+		}
```

Comment:
Not necessary. Only the last `SetTopic` in a message is meaningful.

Reply:
Hmmm did you not say if there is a SetTopic at the end of the message, we should look for it? https://github.com/vgeddes/polkadot-sdk/pull/32#discussion_r1982931679 Or did you just mean the instructions the `converter` inserts after SetTopic?

Reply:
to clarify, we should just ignore any `SetTopic` instructions in `remote_xcm` that are not the last instruction. They are not meaningful.

Reply:
Its a design flaw in xcm.